packages/memory-storage/src/resource-clients/request-queue.ts

```diff
@@ -329,6 +329,7 @@ export class RequestQueueClient extends BaseClient implements storage.RequestQue
         internalRequest.orderNo = forefront ? -unlockTimestamp : unlockTimestamp;
 
         await request?.update(internalRequest);
+        if (forefront) this.forefrontRequestIds.push(id);
```

Comment:
This is just an approximation - on Platform, the `prolongRequestLock` `forefront` order is based on the lock expiration order, not the `prolong` call order. This means that:

```text
prolong lock on A, 10 seconds, forefront
prolong lock on B, 5 seconds, forefront
```

...will result in `B | A | ...tail` in our implementation but `A | B | ...tail` on Platform.

This feature didn't work at all before this PR - this tells me that we don't need to get this detail completely right. Especially since supporting this would require storing the unlock times in the `forefront` list and would be a proper mess maintaining with more `prolong` and `delete` calls.